src/server/schemas/wallet/index.ts

```diff
@@ -20,8 +21,25 @@ export const walletWithAAHeaderSchema = Type.Object({
     ...AddressSchema,
     description: "Smart account address",
   }),
+  "x-account-factory-address": Type.Optional({
+    ...AddressSchema,
+    description: "Smart account factory address",
+  }),
 });
 
+/*
+ * Helper function to parse an address string.
+ * Returns undefined if the address is invalid.
+ */
+export function maybeAddress(address: string | undefined) {
```

Comment:
IMO this function belongs in the primitiveTypes.ts utils file which contain pure functions that do common things like this.

Reply:
ty, wasn't sure of where to put it

Reply:
now that it throws a custom error, not sure if it belongs there anymore?